### ISSUE TITLE
Unwanted copy of Email when moving to a submenu

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/QuickFolders#312]
since v 5.13 Thunderbird makes and unwanted copy of a moved mail (with empty subject line) when we move a mail into a subfolder of a dropdown menu, e.g.:

![image](https://user-images.githubusercontent.com/6314674/192550223-ba30551a-98c8-46a9-8749-6695819161db.png)

this is possibly a regression of changes in #308 but also appears to affect POP accounts.

**Comment by RealRaven2000** [comment]
I have enabled the debug setting extensions.quickfolders.debug.dnd = true (by right-clicking the debug mode checkbox, and it looks like there is a duplication of the drop event when the message is dropped on a submenu as described above:

```
QuickFolders {DND} 16:17:39.981  [3444 ms]   
 buttonDragObserver.drop flavour=text/x-moz-message [quickfolders-util.js:1263:13](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders 16:17:39.981  [0 ms]   
 dragSession =  
XPCWrappedNative_NoHelper { QueryInterface: QueryInterface(), invokeDragSession: invokeDragSession(), getCurrentSession: getCurrentSession(), startDragSession: startDragSession(), startDragSessionForTests: startDragSessionForTests(), endDragSession: endDragSession(), suppress: suppress(), unsuppress: unsuppress(), DRAGDROP_ACTION_NONE: 0, DRAGDROP_ACTION_COPY: 1, … }
[quickfolders-util.js:1268:13](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders {DND} 16:17:39.983  [2 ms]   
 dataTransfer [0] imap-message://thunderbirddaily%40gmail.com@imap.gmail.com/INBOX#2988
[quickfolders-util.js:1263:13](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders {DND} 16:17:39.983  [0 ms]   
 drop: 1 messageUris to mailbox://nobody@Local%20Folders/Local1/Dev/QuickFolders/test1 [quickfolders-util.js:1263:13](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders {DND,QUICKMOVE} 16:17:39.983  [0 ms]   
 QuickFolders.Util.moveMessages: target = test1, makeCopy=false [quickfolders-util.js:1263:13](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders 16:17:39.983  [0 ms]   
 moveMessages  with readStatus = 0 [quickfolders-util.js:1268:13](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders.Util.moveMessages()
  calling CopyMessages (
sourceFolder = Posteingang ,
messages = [xpconnect wrapped nsIMsgDBHdr] ,
destinationFolder = test1 ,
isMove = true 
listener = [object Object] ,
window = [xpconnect wrapped nsIMsgWindow] ,
allowUndo = true) [quickfolders-util.js:866:17](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders {DND} 16:17:39.986  [3 ms]   
 buttonDragObserver.drop flavour=text/x-moz-message [quickfolders-util.js:1263:13](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders 16:17:39.986  [0 ms]   
 dragSession =  
XPCWrappedNative_NoHelper { QueryInterface: QueryInterface(), invokeDragSession: invokeDragSession(), getCurrentSession: getCurrentSession(), startDragSession: startDragSession(), startDragSessionForTests: startDragSessionForTests(), endDragSession: endDragSession(), suppress: suppress(), unsuppress: unsuppress(), DRAGDROP_ACTION_NONE: 0, DRAGDROP_ACTION_COPY: 1, … }
[quickfolders-util.js:1268:13](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders {DND} 16:17:39.987  [1 ms]   
 dataTransfer [0] imap-message://thunderbirddaily%40gmail.com@imap.gmail.com/INBOX#2988
[quickfolders-util.js:1263:13](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders {DND} 16:17:39.988  [1 ms]   
 drop: 1 messageUris to mailbox://nobody@Local%20Folders/Local1/Dev/QuickFolders/test1 [quickfolders-util.js:1263:13](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders {DND,QUICKMOVE} 16:17:39.988  [0 ms]   
 QuickFolders.Util.moveMessages: target = test1, makeCopy=false [quickfolders-util.js:1263:13](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders 16:17:39.988  [0 ms]   
 moveMessages  with readStatus = 0 [quickfolders-util.js:1268:13](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders.Util.moveMessages()
  calling CopyMessages (
sourceFolder = Posteingang ,
messages = [xpconnect wrapped nsIMsgDBHdr] ,
destinationFolder = test1 ,
isMove = true 
listener = [object Object] ,
window = [xpconnect wrapped nsIMsgWindow] ,
allowUndo = true) [quickfolders-util.js:866:17](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders 16:17:39.989  [1 ms]   
 toolbarDragObserver.drop() - dragSession =  
XPCWrappedNative_NoHelper { QueryInterface: QueryInterface(), invokeDragSession: invokeDragSession(), getCurrentSession: getCurrentSession(), startDragSession: startDragSession(), startDragSessionForTests: startDragSessionForTests(), endDragSession: endDragSession(), suppress: suppress(), unsuppress: unsuppress(), DRAGDROP_ACTION_NONE: 0, DRAGDROP_ACTION_COPY: 1, … }
[quickfolders-util.js:1268:13](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders {DND} 16:17:39.989  [0 ms]   
 toolbarDragObserver.drop - text/x-moz-message [quickfolders-util.js:1263:13](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders {DND} 16:17:39.989  [0 ms]   
 toolbarDragObserver.drop text/x-moz-message [quickfolders-util.js:1263:13](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders 16:17:40.21  [32 ms]   
 setFolderUpdateTimer(item):Posteingang [quickfolders-util.js:1268:13](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders 16:17:40.21  [0 ms]   
 Folder Tab Select Timer ID: 376 [quickfolders-util.js:1268:13](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders 16:17:40.22  [1 ms]   
 Setting Update Timer (after timer 371 expired), new Timer: 376 [quickfolders-util.js:1268:13](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders 16:17:40.40  [18 ms]   
 util.touch(test1)
1664291860
27/9 16:17:40 [quickfolders-util.js:1268:13](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders 16:17:40.42  [2 ms]   
 util.touch(test1)
1664291860
27/9 16:17:40 [quickfolders-util.js:1268:13](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders 16:17:40.45  [3 ms]   
 util.touch(test1)
1664291860
27/9 16:17:40 [quickfolders-util.js:1268:13](chrome://quickfolders/content/quickfolders-util.js)
QuickFolders 16:17:40.45  [0 ms]   
 buttonDragObserver.drop DropTarget = menuitem0  Target Folder:test1
```

**Comment by RealRaven2000** [comment]
To avoid retriggering the drop event, I am now calling `evt.preventDefault();` and `evt.stopPropagation(); `earlier (before doing any further processing)

[QuickFolders-mx-5.13.1pre4.zip](https://github.com/RealRaven2000/QuickFolders/files/9657734/QuickFolders-mx-5.13.1pre4.zip)

To install this test version, download the zip file and drag it into Thunderbird Add-ons Manager (do not extract contents, because that won't install).

**Comment by bsb-gh** [comment]
I have just installed the test version, and so far I am not having this issue anymore. I will continue to test.

**Comment by RealRaven2000** [comment]
> I have just installed the test version, and so far I am not having this issue anymore. I will continue to test.

Thanks for testing, can you make sure that the other methods of moving (directly to a tab, or using the quickMove button) also work as expected? I think there are basically 3 cases affected

- moving / copying mails to a QF tab
- moving mails to a subfolder listed directly under a QF tab
- moving mails to more nested subfolders coming of any secondary popup menus that being opened from the first popup menu
- moving new folders to the Toolbar to create tabs (I also use `evt.preventDefault();` and `evt.stopPropagation();` earlier in this version) or add them to a QuickFolders category
- dragging QuickFolders tabs to reorder them:
![image](https://user-images.githubusercontent.com/6314674/192604014-88f7cfc2-8411-41e9-b996-d596f4a8f45a.png)

**Comment by bsb-gh** [comment]
Hello,

All of the tests below have passed for me with no issues.

Brad

**Comment by mtdfh** [comment]
Now the test-version runs on my pc with thunderbird and the problems are solved.
Thanks for your perfect support

Marco

**Comment by RealRaven2000** [comment]
> Now the test-version runs on my pc with thunderbird and the problems are solved. Thanks for your perfect support
> 
> Marco

Thanks for testing to @bsb-gh and @mtdfh , going to upload the new version to ATN soon, already translated the change log into the other 21 languages....

**Comment by RealRaven2000** [comment]
Implemented in 5.13.1 released Sept. 27, 2022
